test/customerBookingsTest.js

```diff
@@ -0,0 +1,86 @@
+import chai from 'chai';
+const expect = chai.expect;
+
+import { sampleData } from '../test/sampleData.js';
+import { getCustomerBookings, calculateTotalCost } from '../src/customerBookings.js'
+
+describe('getCustomerBookings', () => {
+  it('should return all bookings for a customer with multiple bookings', () => {
+    const customerId = 1;
+    const expectedBookings = [
+      { "id": "5fwrgu4i7k55hl6sz", "userID": 1, "date": "2022/04/22", "roomNumber": 1 },
+      { "id": "5fwrgu4i7k55hl6t8", "userID": 1, "date": "2023/01/25", "roomNumber": 1 }
+    ];
+    const customerBookings = getCustomerBookings(customerId, sampleData.bookings);
+    expect(customerBookings).to.deep.equal(expectedBookings);
+  });
+
+  it('should return a single booking for a customer with one booking', () => {
+    const customerId = 5;
+    const expectedBookings = [
+      {"id": "5fwrgu4i7k55hl6t7", "userID": 5, "date": "2023/01/19", "roomNumber": 5}
+    ];
+    const customerBookings = getCustomerBookings(customerId, sampleData.bookings);
+    expect(customerBookings).to.deep.equal(expectedBookings);
+  });
+
+  it('should return an empty array for a customer with no bookings', () => {
+    const customerId = 55;
+    const customerBookings = getCustomerBookings(customerId, sampleData.bookings);
+    expect(customerBookings).to.be.an('array').that.is.empty;
+  });
+
+  it('should handle invalid customer ID types', () => {
```

Comment:
I'm not sure I understand why this test is necessary. Is it for the extension managerial view? Because in theory the non-extension version of the project will have a user login and therefore the customer ID will always be valid. Something to consider.

Reply:
I am reaching for more sad paths and probably forced it in my search for something.